src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs

```diff
@@ -274,11 +274,24 @@ private async Task<List<CloudApplication>> GetExistingBeanstalkEnvironments()
                 var manifestPath = Path.Combine(Path.GetDirectoryName(projectPath) ?? string.Empty, Constants.ElasticBeanstalk.WindowsManifestName);
                 if (_fileManager.Exists(manifestPath))
                 {
-                    var manifest = JsonSerializer.Deserialize<ElasticBeanstalkWindowsManifest>(await _fileManager.ReadAllTextAsync(manifestPath));
-                    if (manifest.Deployments.AspNetCoreWeb.Count != 0)
+                    try
                     {
-                        optionSettings[Constants.ElasticBeanstalk.IISWebSiteOptionId] = manifest.Deployments.AspNetCoreWeb[0].Parameters.IISWebSite;
-                        optionSettings[Constants.ElasticBeanstalk.IISAppPathOptionId] = manifest.Deployments.AspNetCoreWeb[0].Parameters.IISPath;
+                        var manifest = JsonSerializer.Deserialize<ElasticBeanstalkWindowsManifest>(await _fileManager.ReadAllTextAsync(manifestPath), new JsonSerializerOptions
+                        {
+                            ReadCommentHandling = JsonCommentHandling.Skip
+                        });
+                        if (manifest.Deployments.AspNetCoreWeb.Count != 0)
+                        {
+                            optionSettings[Constants.ElasticBeanstalk.IISWebSiteOptionId] = manifest.Deployments.AspNetCoreWeb[0].Parameters.IISWebSite;
+                            optionSettings[Constants.ElasticBeanstalk.IISAppPathOptionId] = manifest.Deployments.AspNetCoreWeb[0].Parameters.IISPath;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new InvalidWindowsManifestFileException(
```

Comment:
This exception seems like it is only applicable to the Deserialize call - I would imagine that any exception arising after Deserialize would be a different problem. Should the deserialize call and the try-catch that raises InvalidWindowsManifestFileException be moved to a dedicated function?

Reply:
I separated the IO into a different function.